lib/chef/knife/ec_base.rb

```diff
@@ -160,7 +164,8 @@ def users_for_purge
       end
 
       def knife_ec_error_handler
-        @knife_ec_error_handler ||= Chef::Knife::EcErrorHandler.new(dest_dir, self.class)
+        error_dir = !config[:error_log_dir].nil? ? config[:error_log_dir] : dest_dir
```

Comment:
If you're just trying to use error_log_dir unless it's nil couldn't this just be:

```suggestion
        error_dir = config[:error_log_dir] || dest_dir
```

Reply:
Super! My ruby is getting rusty. Thanks.
https://github.com/chef/knife-ec-backup/pull/135/commits/c450b1bf4cd31edc94986081b09a2a95ba827bd5